jenkins_node_scanner.py

```diff
@@ -224,30 +243,22 @@ def main():
     while True:
         with ignore_exceptions_except_exit():
             with GLOBAL_EXCEPTIONS.count_exceptions():
-                logging.debug('Fetching node list')
-                nodes = [x for x in get_nodes(master)
-                         if not exclude_regex.match(x['name'])]
-                logging.debug('Found %d nodes', len(nodes))
-
-                raw_node_ips = [(node, get_node_ip(master, node)) for node in nodes]
-                node_ips = [(node, ip) for node, ip in raw_node_ips if ip is not None]
-
-                # pylint: disable=no-member
-                NODES_FOUND.labels(args.url).set(len(node_ips))
-                # pylint: disable=no-member
-                UNPARSEABLE_NODES_FOUND.labels(args.url).set(
-                    len(raw_node_ips) - len(node_ips))
-
-                node_info = [{
-                    'labels': {
-                        'jenkins_master': args.url,
-                        'node': node['name'],
-                    },
-                    'targets': ['%s:%d' % (ip_addr, port) for port in args.target_ports
-                                if is_port_open(ip_addr, port)],
-                } for node, ip_addr in node_ips]
-
-                write_output(args.output_file, node_info)
+                node_infos = []
+
+                for url in args.urls:
+                    logging.debug('Connecting to Jenkins master at %s', url)
+                    master = jenkins.Jenkins(
+                        url,
+                        username=args.username,
+                        password=args.password,
+                        timeout=args.timeout,
+                    )
+
+                    node_infos.append(
```

Comment:
Shouldn't this be `node_infos.extend(...)`?

Reply:
Good catch, fixed in 475a696f6cdb387c69ad2323e1c947dcb85464f9

Reply:
Then I guess `get_node_info` could have a better name.